src/libmongoc/tests/unified/runner.c

```diff
@@ -1585,10 +1588,11 @@ test_install_unified (TestSuite *suite)
 {
    run_unified_tests (suite, JSON_DIR "/unified");
 
-   /* crud/unified spec tests also use the unified format. */
    run_unified_tests (suite, JSON_DIR "/crud/unified");
 
    run_unified_tests (suite, JSON_DIR "/transactions/unified");
 
    run_unified_tests (suite, JSON_DIR "/collection-management");
+
+   run_unified_tests (suite, JSON_DIR "/sessions/unified");
```

Comment:
Just a heads up, I've deviated from this already for the Versioned API, as I thought it'd be more sensible to add the unified spec tests for a spec in the same place where we would've added the legacy spec tests. As far as I can tell, this wouldn't affect the `SKIPPED_TESTS` behaviour, but it would keep tests for one spec in one place. No strong preference on this though, and I can move the Versioned API tests if we decide that this is the correct place to add new unified tests.

Reply:
Ah I see, I'll leave it up to @kevinAlbs. The Go and CXX drivers do keep all the unified spec tests in one runner as is done here. Either way, I'll either move this to `test-mongoc-client-session` or move the versioned API tests here.

Reply:
I think we should match the directory structure in the specifications repo. Keeping legacy and unified in separate directories makes sense since they go through different runners.

DRIVERS-1607 moves the existing sessions tests into:

`source/sessions/tests` => `source/sessions/tests/legacy`

And the new unified tests are put into:

`source/sessions/tests/unified`

I think that should correspond to these directories in the C driver (as proposed in this PR):

- `src/libmongoc/tests/json/sessions/legacy`
- `src/libmongoc/tests/json/sessions/unified`

@alcaeus Are you referring to the Versioned API tests in `src/libmongoc/tests/json/versioned_api`? That matches the specifications repo since there is no distinction between legacy and unified.

Reply:
Unless I'm mistaken, @alcaeus was referring to where the tests are run from, not where the `json` files are. In this case, I've added `run_unified_tests` here in `runner.c`, whereas the versioned API tests (which are unified) are run from [`test-mongoc-versioned-api`](https://github.com/mongodb/mongo-c-driver/blob/master/src/libmongoc/tests/test-mongoc-versioned-api.c#L179).

Reply:
Gotcha, since both the `/sessions/unified` and `/versioned-api` tests are grep-able paths, I'll just leave them where they both are unless someone has a strong preference.

Reply:
No preference here, I'll clean up the other tests when I get a chance